SCClassLibrary/Common/Core/Clock.sc

```diff
@@ -327,6 +327,48 @@ elapsed time is whatever the system clock says it is right now. elapsed time is
 		^this.beats - this.bars2beats(this.bar)
 	}
 
+	sync { arg tempo, secs = 4, resolution = 1;
+		var next, time, durCur, durNew, durDif, durAvg, stepsPerBeat,
+			delta, factor, steps, stepSize, sum, lastDur, index = 0;
+
+		secs = secs.max(0.03);				// saftey and lower jitter limit
+		next = this.timeToNextBeat(1);
+		time = secs - (this.tempo.reciprocal * next);
+		if(time < next) {						// jump directly
+			this.tempo_(next / secs);		// set a high tempo
+			this.sched(next, { this.tempo_(tempo); nil });
+		} {							// else interpolate
+			this.sched(next, {		// start at next beat
+				durCur = this.tempo.reciprocal;
```

Comment:
These variables need clearer names, and I think this block could definitely do with a good once-over to make sure all these variables are really necessary and to get the code in a more readable state. For instance, `delta` is assigned 12 lines before it's used, and interrupts the flow of calculations. It also looks like `delta`, `factor`, and `stepSize` can probably be eliminated since they just store the result of small math ops and are only used once or twice after that.

In addition, I would recommend adding parentheses to clearly show order of operations, especially when relying on SC's lack of operator precedence (lines 345, 350, 351).